### PR TITLE
Add TC_CCTRL_2_1, TC_CCTRL_2_2, TC_CCTRL_2_3 to CI

### LINKED ISSUE #35484: [Workflow] Run CCTRL test in CI
### Workflow issues

Integrate the following FS-related tests into CI:

- [x] CCTRL_2_1
- [x] CCTRL_2_2
- [x] CCTRL_2_3

### Platform

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by arkq**
### Problem

Fixes #35484

### Changes

- Allow to override test runner YAML options with `scripts/tests/run_python_test.py` command line options
- Add TC_CCTRL_2_1, TC_CCTRL_2_2, TC_CCTRL_2_3 to CI

### Testing

CI will check

### CODE REVIEW

File: src/app/tests/suites/certification/ci-pics-values [hunk 1]
```diff
@@ -922,6 +922,9 @@ MCORE.BDX.Sender=0
 MCORE.BDX.SynchronousReceiver=0
 MCORE.BDX.SynchronousSender=0
 
+# Fabric Synchronization
+MCORE.FS=1
```
Comment (yufengwangca) [inline src/app/tests/suites/certification/ci-pics-values:926]: This enable the FS feature by default in SDK, we have passed the SVE certs, but need the certification team to approve this feature in the spec

**Review by yufengwangca — APPROVED** [review]

File: src/app/tests/suites/certification/ci-pics-values (hunk 1 above)
Comment (yufengwangca) [reply src/app/tests/suites/certification/ci-pics-values:926]: I just built the ToT for the Matter spec and noticed that the FS is already included in the ToT. I believe we can enable this feature now and don’t need to wait for the official 1.4 spec. I'll submit another PR to remove the provisional status of the XML.

**Review by saurabhst — APPROVED** [review]

File: src/python_testing/TC_CCTRL_2_1.py [hunk 2]
```diff
@@ -15,6 +15,28 @@
...
+#     script-args: >
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --endpoint 0
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factoryreset: true
```
[the remark this answers is not in the document]
Comment (arkq) [reply src/python_testing/TC_CCTRL_2_1.py:36]: Yes, I'm planing to updating everything to "factory-reset" during a YAML-conversion pass (before dropping non-YAML format).

File: src/python_testing/matter_testing_infrastructure/chip/testing/metadata.py [hunk 3]
```diff
@@ -40,7 +40,7 @@ class Metadata:
     factory_reset: bool = False
     factory_reset_app_only: bool = False
     script_gdb: bool = False
-    quiet: bool = True
+    quiet: bool = False
```
[the remark this answers is not in the document]
Comment (arkq) [reply src/python_testing/matter_testing_infrastructure/chip/testing/metadata.py:43]: Having quiet by default is kind of non-practical. During development it's nice to have output. Anyway, all tests have `quiet: true` in them. Also, if quiet is the default, why we do not have "verbose" option, to force tests to print something? This reasoning was behind changing this default. I'm always trying to follow the rule of "working out of the box with sane build-in defaults".

**Pull request merged**